### PR TITLE
[CASSANDRA-18340] Upgrade snakeyaml from 1.26 to 2.0 for CVE's 

### PR DESCRIPTION
**Pull request by bipinprasad**
CVE's are listed here https://mvnrepository.com/artifact/org.yaml/snakeyaml/1.26

Constructor signataure has changed in the snakeyaml 2.0. Apache Storm project has an example project "storm-cassandra".
To fully implement snakeyaml 2,0 in Storm, it needs a Cassandra build with smakeyaml 2.0 as well.
 
<Optional lengthier description (context on patch)>

patch by <bipinprasad@apache.org>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/browse/CASSANDRA-18340)

### CODE REVIEW

File: src/java/org/apache/cassandra/utils/concurrent/AbstractFuture.java [hunk 1]
```diff
@@ -494,11 +494,11 @@ public Future<V> awaitThrowUncheckedOnInterrupt() throws UncheckedInterruptedExc
     public String toString()
     {
         String description = description();
-        String state = state();
+        String state = mystate();
```
Comment (smiklosovic) [inline src/java/org/apache/cassandra/utils/concurrent/AbstractFuture.java]: why this change?

Comment (bipinprasad) [reply src/java/org/apache/cassandra/utils/concurrent/AbstractFuture.java]: Conflict with same name but different signature (public State state()) method in Future.

**Comment by ivo2d** [comment]
@smiklosovic @bipinprasad any chance this can merge into 3.11 as it is critical and it has been in progress for a couple of months? I know you have stopped security patching a week ago, but if you can please do it, I would appreciate it. :)

**Comment by bipinprasad** [comment]
Rebased. 
Need approval after the rebase.

**Comment by bipinprasad** [comment]
Created new PR https://github.com/apache/cassandra/pull/2455

**Comment by ddezoysa** [comment]
Can someone please share in which release this snakeyaml upgrade to 2.0 fix supposed to be released?
